File: enemyAi.js
var Distance;
var Target:Transform;
var attackrange=2.0;
var theDamage=40;
private var attackTime:float;
var attackRepeatTime=1;
var controller:CharacterController;

function Start () {
	attckTime=Time.time;
}

function Update () {
	Distance=Vector3.Distance(Target.position,transform.position);
	if(Distance<=attackrange)
	{
	AttackPlayer();
	}
}
function AttackPlayer()
{
if(Time.time>attackTime)
{
Target.SendMessage("ApplyDamage",theDamage);
attackTime=Time.time+attackRepeatTime;
}
}

File: enemyAi.cs


using System.Collections;
using System.Collections.Generic;
using UnityEngine;

public class enemyAi : MonoBehaviour {
	float distance;
	public Transform target;
	public float attackrange=2;
	public float damage=40;
	private float AttackTime;
	int AttackRepeatTime=1;
	public CharacterController character;


	// Use this for initialization
	void Start () {
		AttackTime = Time.time;
		//ph = GameObject.Find ("FPSController").GetComponent<player_health> ();


	}
	
	// Update is called once per frame
	void Update () {
		distance = Vector3.Distance (target.position, this.transform.position);
		if (distance <= attackrange) 
		{
			AttackPlayer ();
		}
	}
	void AttackPlayer()
	{
		
		if (Time.time > AttackTime) 
		{
			//ph.ApplyDamage (40);
			target.BroadcastMessage ("ApplyDamage", damage);
			AttackTime = Time.time + AttackRepeatTime;

		}	
	}
}
